translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -9441,15 +9478,155 @@ Value *SPIRVToLLVM::transGLSLExtInst(SPIRVExtInst *ExtInst,
     // Inverse of square matrix
     return getBuilder()->CreateMatrixInverse(Args[0]);
 
+  case GLSLstd450Modf: {
+    // Split input into fractional and whole number parts.
+    Value *WholeNum =
+        getBuilder()->CreateUnaryIntrinsic(Intrinsic::trunc, Args[0]);
+    Value *Fract = getBuilder()->CreateFSub(Args[0], WholeNum);
+    getBuilder()->CreateStore(WholeNum, Args[1]);
+    return Fract;
+  }
+
+  case GLSLstd450ModfStruct: {
+    // Split input into fractional and whole number parts.
+    Value *WholeNum =
+        getBuilder()->CreateUnaryIntrinsic(Intrinsic::trunc, Args[0]);
+    Value *Fract = getBuilder()->CreateFSub(Args[0], WholeNum);
+    Value *Result = UndefValue::get(transType(ExtInst->getType()));
+    Result = getBuilder()->CreateInsertValue(Result, Fract, 0);
+    Result = getBuilder()->CreateInsertValue(Result, WholeNum, 1);
+    return Result;
+  }
+
+  case GLSLstd450FMin:
+  case GLSLstd450NMin: {
```

Comment:
we need support nan for NMin

Reply:
Good spot. I had already found that in a CTS failure and fixed it locally, but I forgot to fix nmax in the same way. :-)